src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -72,6 +72,7 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
             }
 
             // Get the workspaces.
+            executionContext.Output(StringUtil.Loc("QueryingWorkspaceInfo"));
```

Comment:
Querying can take a while. This is an enhancement that has been in the back of my mind.